### PR TITLE
"Pick in Store" button always visible on checkout page...

### PR DESCRIPTION
**Pull request by nmalevanec**
### Fixed Issues (if relevant)

1. https://github.com/magento/inventory/issues/2647: "Pick in Store" button always visible on checkout page.
2. https://github.com/magento/inventory/issues/2650: Add to configuration of Store pickup method radius for search results of store selection

### Manual testing scenarios (*)

1. see https://github.com/magento/inventory/issues/2647

### For search radius
1. Create two sources with store pickup.(Culver City, Austin)
2. Create simple product and assign two sources to product.
3. Add product to cart and navigate to checkout.
4. Click 'Pick in Store' try to search Culver City. 
5. Verify, only Culver City source found.
6. Navigate to admin area Store>Configuration>Sales>Delivery Methods>In-Store Delivery
7. Change search radius to 2000 or greater
8. Navigate to checkout.
9. Click 'Pick in Store' try to search Culver City. 
10. Verify, both sources for Culver City and Austin are found.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by sidolov** [comment]
@magento run all tests

File: app/code/Magento/InventoryInStorePickupFrontend/Block/Checkout/LayoutProcessor.php [hunk 1]
```diff
@@ -0,0 +1,204 @@
+<?php
```
Comment (sidolov) [inline app/code/Magento/InventoryInStorePickupFrontend/Block/Checkout/LayoutProcessor.php:1]: Please, revert checkout layout xml and add only missing data in the LayoutProcessor

File: app/code/Magento/InventoryInStorePickupFrontend/Model/Validator/StorePickUpValidator.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+use Magento\InventoryInStorePickupApi\Api\GetPickupLocationsInterface;
+use Magento\InventoryInStorePickupApi\Model\SearchRequestBuilderInterface;
+use Magento\Store\Model\ScopeInterface;
+
+/**
+ * Verify, if store pick-up available for given website.
+ */
+class StorePickUpValidator
+{
+    const CONFIG_PATH = 'carriers/in_store/active';
```
Comment (sidolov) [inline app/code/Magento/InventoryInStorePickupFrontend/Model/Validator/StorePickUpValidator.php]: Please, make const private

File: app/code/Magento/InventoryInStorePickupFrontend/view/frontend/web/js/view/store-selector.js [hunk 3]
```diff
@@ -163,7 +163,13 @@ define([
          * @returns {*}
          */
         updateNearbyLocations: function (address) {
-            var self = this;
+            var self = this,
+                skus = [],
+                items = quote.getItems();
+
+            _.each(items, function (item) {
```
Comment (sidolov) [inline app/code/Magento/InventoryInStorePickupFrontend/view/frontend/web/js/view/store-selector.js]: Will it work with dynamic skus? For example configurable product with custom options?

File: app/code/Magento/InventoryInStorePickupFrontend/Model/Validator/StorePickUpValidator.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Validation\ValidationResultFactory;
+use Magento\InventoryInStorePickupApi\Api\GetPickupLocationsInterface;
+use Magento\InventoryInStorePickupApi\Model\SearchRequestBuilderInterface;
+use Magento\Store\Model\ScopeInterface;
+
+/**
+ * Verify, if store pick-up available for given website.
+ */
+class StorePickUpValidator
```
Comment (sidolov) [inline app/code/Magento/InventoryInStorePickupFrontend/Model/Validator/StorePickUpValidator.php]: What exactly this class validates? Could you please change to the name more explicit?

**Review by sidolov — CHANGES_REQUESTED** [review]

File: app/code/Magento/InventoryInStorePickupFrontend/Block/Checkout/LayoutProcessor.php (hunk 1 above)
Comment (nmalevanec) [reply app/code/Magento/InventoryInStorePickupFrontend/Block/Checkout/LayoutProcessor.php:1]: Done.

File: app/code/Magento/InventoryInStorePickupFrontend/Model/Validator/StorePickUpValidator.php (hunk 2 above)
Comment (nmalevanec) [reply app/code/Magento/InventoryInStorePickupFrontend/Model/Validator/StorePickUpValidator.php]: Done.

File: app/code/Magento/InventoryInStorePickupFrontend/view/frontend/web/js/view/store-selector.js (hunk 3 above)
Comment (nmalevanec) [reply app/code/Magento/InventoryInStorePickupFrontend/view/frontend/web/js/view/store-selector.js]: Fixed

File: app/code/Magento/InventoryInStorePickupFrontend/Model/Validator/StorePickUpValidator.php (hunk 4 above)
Comment (nmalevanec) [reply app/code/Magento/InventoryInStorePickupFrontend/Model/Validator/StorePickUpValidator.php]: Renamed

**Review by sidolov — APPROVED** [review]

**Comment by sidolov** [comment]
@magento run Functional Tests CE

**Comment by sidolov** [comment]
@magento run Unit Tests

**Comment by sidolov** [comment]
@magento run Functional Tests CE,Unit Tests

**Pull request merged**